### PR TITLE
feat: AutoQASM public repo setup

### PR DESCRIPTION
**Pull request by rmshaffer**
*Issue #, if available:*

*Description of changes:*
Variety of things to cleanup the standalone AutoQASM public repo.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/scripts/update_dependency.py [hunk 1]
```diff
@@ -14,21 +14,19 @@
 import fileinput
 from pathlib import Path
 
-# Here we replace the `amazon-braket-sdk` dependency to point to the file system; otherwise
+# Here we replace the `autoqasm` dependency to point to the file system; otherwise
 # pip will install them separately, allowing it to override the version of
-# any mutual dependencies with the sdk. By pointing to the file system, pip will be
-# forced to reconcile the dependencies in setup.py with the dependencies of the sdk,
+# any mutual dependencies with autoqasm. By pointing to the file system, pip will be
```
Comment (laurencap) [inline .github/scripts/update_dependency.py:19]: Capitalization should be AutoQASM, right?

**Review by laurencap — DISMISSED** [review]

**Comment by laurencap** [comment]
The docs here: https://autoqasm.readthedocs.io/en/latest/_apidoc/autoqasm.html are missing code formatting around the code example

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -186,7 +186,7 @@ If a parameter of a function has a default value, please note what the default i
 If that default value is `None`, it can also be helpful to explain what happens when the parameter is `None`.
 If `**kwargs` is part of the function signature, link to the parent class(es) or method(s) so that the reader knows where to find the available parameters.
 
-For an example file with docstrings, see [the `circuit` module](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/src/braket/circuits/circuit.py).
+For an example file with docstrings, see [the `gates` module](https://github.com/amazon-braket/autoqasm/blob/main/src/autoqasm/instructions/gates.py).
```
Comment (yitchen-tim) [inline CONTRIBUTING.md:189]: The docstrings for `gates` are pretty minimal. Maybe we can use [`measurements` ](https://github.com/amazon-braket/autoqasm/blob/main/src/autoqasm/instructions/measurements.py) which also has a code example in module level doc and docs for returns.

Comment (rmshaffer) [reply CONTRIBUTING.md:189]: That's a good idea, thank you.

**Comment by rmshaffer** [comment]
> The docs here: https://autoqasm.readthedocs.io/en/latest/_apidoc/autoqasm.html are missing code formatting around the code example

Good catch @laurencap, thank you! Added `code-block` directives to fix this.

**Pull request merged**